radix-common/src/crypto/ed25519/private_key.rs

```diff
@@ -1,28 +1,22 @@
 use super::Ed25519Signature;
 use crate::internal_prelude::*;
-use ed25519_dalek::{Keypair, PublicKey, SecretKey, Signer};
-use zeroize::Zeroize;
+use ed25519_dalek::{Signer, SigningKey};
 
-#[derive(Zeroize)]
-#[zeroize(drop)]
-pub struct Ed25519PrivateKey(SecretKey);
+// No need to derive Zeroize.
```

Comment:
I believe we still need explicit `zeroize` to allow clients to zeroize without dropping - e.g. I think Sargon needed it?

Reply:
https://docs.rs/zeroize/latest/zeroize/
> This crate has custom derive support for the Zeroize trait, gated under the zeroize crate’s zeroize_derive Cargo feature, which automatically calls zeroize() on all members of a struct or tuple struct.

Reply:
I guess we technically don't need "on drop" if the child is doing it on drop already

Reply:
It's mainly needed for zeroize composition, especially considering that one can't extract the inner contents.

Reply:
Unfortunately it is not possible to derive Zeroize for `SigningKey`, because only `secret_key` implements `zeroize()` and `verifying_key` does not.
```
pub struct SigningKey {
    /// The secret half of this signing key.
    pub(crate) secret_key: SecretKey,   
    /// The public half of this signing key.
    pub(crate) verifying_key: VerifyingKey,
}
```
We also does not have access to the `SigningKey` fields, so we could implement zeroize manually.

So I see 2 options:
1. back to the previous implementation, where `Ed25519PrivateKey` was wrapping `SecretKey` and derive Zeroize
```
pub struct Ed25519PrivateKey2(SecretKey)
```
2.  Use unsafe manual zeroing.

I think we should rather avoid unsafe, so'd prefer option 1.

WDYT

Reply:
Done.
slack discussion [here](https://rdxworks.slack.com/archives/C071C6DB8TG/p1727867648027169?thread_ts=1727706454.898939&cid=C071C6DB8TG)